core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -423,4 +435,76 @@ public void run() {
       }
     }
   }
+
+  public static final Metadata.Key<Long> TIMEOUT_KEY =
```

Comment:
And that's because we don't have a server-api yet...

Let's add a TODO saying that this is only temporarily public and a JavaDoc comment that it is intended for internal use only.